ci/base-images/cdxgen/Dockerfile.dotnet10

```diff
@@ -0,0 +1,28 @@
+FROM ghcr.io/cyclonedx/bci-dotnet10:master
```

Comment:
This must be `FROM ghcr.io/cyclonedx/debian-dotnet10:master`. bci is for SLE images, while this PR only includes debian. Also move this file to inside debian directory otherwise the build will fail.